### PR TITLE
initial draft of neurodocker support

### PR DESCRIPTION
**Pull request by PeerHerholz**
- this PR introduces the first draft of a potential neurodocker support
- it changes/adds the following files: `generate_freesurfer_images.sh`, `Dockerfile`, `Singularity`
- `generate_freesurfer_images.sh` is a bash script that utilizes `neurodocker` to create the `Dockerfile` and `Singularity recipe`
- `Dockerfile` is the `Dockerfile` created through `generate_freesurfer_images.sh`
- `Singularity` is the `Singularity` recipe created through `generate_freesurfer_images.sh`

### CODE REVIEW

File: generate_freesurfer_images.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+###############################################################################
+
+image="kaczmarj/neurodocker:master@sha256:936401fe8f677e0d294f688f352cbb643c9693f8de371475de1d593650e42a66"
+
+# Generate a dockerfile for building BIDS-Apps Freesurfer container
+docker run --rm ${image} generate docker \
+  --base ubuntu:xenial \
+  --pkg-manager apt \
+  --install tcsh bc tar libgomp1 perl-modules wget curl \
+    python3 python3-pip python3-pandas python2.7 python-pip \
```
Comment (Shotgunosine) [inline generate_freesurfer_images.sh]: We might not need python 2.7.

**Comment by Shotgunosine** [comment]
Tests are failing because Neurodocker's Freesurfer is looking for a license at `/opt/freesurfer/license.txt`. We could copy the license file to that location in the circlci tests.

**Comment by Shotgunosine** [comment]
Also, it does build for me, I was able to get the first test running with this command:
`docker run -ti --rm --read-only -v ~/license/license.txt:/opt/freesurfer/license.txt --tmpfs /tmp --tmpfs /var/tmp -v ${HOME}/data/ds114_test1:/bids_dataset -v ${HOME}/outputs1:/outputs bids/${CIRCLE_PROJECT_REPONAME} /bids_dataset /outputs participant --participant_label 01 --license_file=/opt/freesurfer/license.txt --stages autorecon1 && cat ${HOME}/outputs1/sub-01/scripts/recon-all.done`

File: generate_freesurfer_images.sh (hunk 1 above)
Comment (PeerHerholz) [reply generate_freesurfer_images.sh]: True that, I'll remove it.

**Comment by PeerHerholz** [comment]
> Tests are failing because Neurodocker's Freesurfer is looking for a license at `/opt/freesurfer/license.txt`. We could copy the license file to that location in the circlci tests.

Yes, sounds about right. Should I do that or do you want to check it? We then would also have to update the `readme`, indicating the specific path the license should be mapped to.

**Comment by Shotgunosine** [comment]
I'll give it a shot, may be able to just use 6.0.1 instead of 6.0.0-min.

**Comment by PeerHerholz** [comment]
> I'll give it a shot, may be able to just use 6.0.1 instead of 6.0.0-min.

AFAIK `neurodocker` currently "only" supports version `6.0.0-min`. Might be worth opening an issue for that in the corresponding repo.

**Comment by Shotgunosine** [comment]
Nah, you can specify 6.0.1, you just don't get a minified version, which is fine.

**Comment by PeerHerholz** [comment]
True that, just checked it. Their `readme` needs a respective update. Will open an issue there.

**Comment by PeerHerholz** [comment]
Looks about right, reopening for review. Thx @Shotgunosine!

**Comment by PeerHerholz** [comment]
Tests are passing and I addressed your comments. Hence, I think we can go ahead and merge, no? @Shotgunosine

**Comment by Shotgunosine** [comment]
Yep, I think so.

**Pull request merged**